Library/Homebrew/os/mac.rb

```diff
@@ -37,6 +37,15 @@ def outdated_release?
       version < "10.9"
     end
 
+    def os_name(version = @version)
+      # TODO: replace os_name by constant string "macOS" when macOS 10.14 is released
+      if Version.create(version) >= Version.create("10.12")
```

Comment:
Like @DomT4 [said before](https://github.com/Homebrew/brew/pull/964#r78884726), plain version comparison doesn't work well as `"10.7" > "10.12"` in the plain string comparison.